protected/models/CurationLog.php

```diff
@@ -81,29 +81,46 @@ public function attributeLabels()
             'last_modified_by' => 'Last Modified By',
         );
     }
-    
-    public static function createlog($status,$id) {
-       
-        $curationlog = new CurationLog;
-        $curationlog->creation_date = date("Y-m-d");
-        $curationlog->last_modified_date = null;
-        $curationlog->dataset_id = $id;
-        $curationlog->created_by = "System";
-        $curationlog->action = "Status changed to ".$status;
-        if (!$curationlog->save())
-            return false;
-    }
-    
-    public static function createlog_assign_curator($id,$creator,$username) {
 
-        $curationlog = new CurationLog;
-        $curationlog->creation_date = date("Y-m-d");
-        $curationlog->last_modified_date = null;
-        $curationlog->dataset_id = $id;
-        $curationlog->created_by = $creator;
-        $curationlog->action = "Curator Assigned"." $username";
-        if (!$curationlog->save())
-            return false;
+//    public static function createLogEntry($id)
+//    {
+//        $curationlog = new CurationLog;
+//        $curationlog->creation_date = date("Y-m-d");
+//        $curationlog->last_modified_date = null;
+//        $curationlog->dataset_id = $id;
+//    }
+
+//    public static function createlog($status,$id) {
+//
+//        self::createLogEntry($id);
+//        $curationlog->created_by = "System";
+//        $curationlog->action = "Status changed to ".$status;
+//        if (!$curationlog->save())
+//            return false;
+//    }
+//
+//    public static function createlog_assign_curator($id,$creator,$username) {
+//
+//        self::createLogEntry($id);
+//        $curationlog->created_by = $creator;
+//        $curationlog->action = "Curator Assigned"." $username";
+//        if (!$curationlog->save())
+//            return false;
+//    }
+
+    public static function createCurationLogEntry($id)
+    {
+        $model = Dataset::model()->findByPk($id);
+        if ($model->upload_status !== "Published") {
+            $curationlog = new CurationLog;
+            $curationlog->creation_date = date("Y-m-d");
+            $curationlog->last_modified_date = null;
+            $curationlog->dataset_id = $id;
+            $curationlog->created_by = "System";
+            $curationlog->action = "Status changed to stuff";
+            if (!$curationlog->save())
+                return false;
+        }
```

Comment:
Hi @rija I tried to extract the most common code from the static functions, but it did not work. Do you have reference/documentation on how to factor out common code?

Reply:
Hi @kencho51 

have a look here:

* https://github.com/rija/gigadb-website/blob/fuw-cicd/protected/models/CurationLog.php
* https://github.com/rija/gigadb-website/blob/fuw-cicd/protected/tests/unit/CurationLogTest.php